internal/controller/bucket/bucket.go

```diff
@@ -186,9 +193,9 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	allBackends := c.backendStore.GetAllBackends()
 	for s3BackendName := range allBackends {
 		go func(backendName, bucketName string) {
-			bucketExists, err := c.bucketExists(context.Background(), backendName, bucketName)
+			bucketExists, err := c.bucketExists(ctx, backendName, bucketName)
```

Comment:
Does it make sense to cancel these requests once we have found the bucket on the first cluster?